rest/route/host_spawn.go

```diff
@@ -71,6 +72,22 @@ func (hph *hostPostHandler) Run(ctx context.Context) gimlet.Responder {
 		hph.options.SetDefaultTimeZone(user.Settings.Timezone)
 	}
 
+	if !user.HasDistroCreatePermission() {
```

Comment:
Could we add a test for this behavior for this route?

Reply:
I was slightly hoping not to because it's a small thing and the existing test for this route are a hot mess, but yeah, it's valid to write a test. I rewrote the tests, so it requires re-review.

Reply:
You are a role model 😆 . Thank you for rewriting them!